legend-engine-xt-relationalStore-grammar/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperRelationalBuilder.java

```diff
@@ -961,6 +963,17 @@ private static PropertyMapping processRelationalPropertyMapping(RelationalProper
             Assert.assertTrue(eMap != null, () -> "Can't find enumeration mapping '" + propertyMapping.enumMappingId + "'");
             res = res._transformer(eMap);
         }
+        try
+        {
+            Root_meta_relational_mapping_assertCompatibleTypesInRelationalPropertyMapping_RelationalPropertyMapping_$0_1$__Boolean_1_(res, context.pureModel.getExecutionSupport());
+        }
+        catch (Exception e)
+        {
+            if (e.getMessage().contains("\"Types of class property: ") && e.getMessage().endsWith(" are incompatible\""))
+            {
+                context.pureModel.addWarnings(Lists.mutable.with(new Warning(SourceInformationHelper.fromM3SourceInformation(res.getSourceInformation()), e.getMessage())));
+            }
+        }
```

Comment:
Do we really need exception handling to add warnings?

Can we just read the result and add warning if check failed (no check required on message as well).

try() still remains to make sure it doesn't break due to other issues